### PR TITLE
python312Packages.triton-bin: 2.1.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by DerDennisOP**
## Description of changes

Only Binary Files, appendix to #328247

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rehno-lindeque** [comment]
Could we add `isPy312` since there's a binary hash for 3.12 now?

**Comment by DerDennisOP** [comment]
@SomeoneSerge is this ready to merge too?

File: pkgs/development/python-modules/triton/binary-hashes.nix [hunk 1]
```diff
@@ -7,26 +7,31 @@
...
+      name = "triton-3.0.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl ";
+      url = "https://files.pythonhosted.org/packages/ea/6b/1d72cc8a7379822dadf050474add7d8b73b02c35057446b6f17d27cb9ea2/triton-3.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl";
+      hash = "sha256-QQBPsa6aU/yz6XB0X+uH8OPJTGzhuobpX6O4U3iUvvc=";
     };
     x86_64-linux-310 = {
-      name = "triton-2.1.0-cp310-cp310-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/triton-2.1.0-0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl";
-      hash = "sha256-ZkOZI6MNXUg5mwip6uEDcPbCYaXshkpkmDuuYxUtOdc=";
+      name = "triton-3.0.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl ";
+      url = "https://files.pythonhosted.org/packages/15/67/84e5a4b7b45bdeb11da26a67dfa2b988c512abbcbcad8cbc30aa579051b2/triton-3.0.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl";
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/triton/binary-hashes.nix:23]: Right, about this 1, as I mentioned in https://github.com/NixOS/nixpkgs/pull/328247 I am concerned about changing provenance. We used to download from one party (pytorch) now we're downloading from another (pypi) which was just recently specifically exploited in the context of pytorch.

Did pytorch stop distributing triton wheels or what's the situation here?

Comment (DerDennisOP) [reply pkgs/development/python-modules/triton/binary-hashes.nix:23]: i think so

Comment (DerDennisOP) [reply pkgs/development/python-modules/triton/binary-hashes.nix:23]: nvm, I just checked, they added the new version now too. I'll edit my commit.

Comment (DerDennisOP) [reply pkgs/development/python-modules/triton/binary-hashes.nix:23]: I'm done with the changes, gonna commit tomorrow.

**Comment by SomeoneSerge** [comment]
Very last nit: can you update the commit message with the up-to-date attribute name? 🙃 mhm maybe I get to do it in the morning myself

**Review by SuperSandro2000 — APPROVED** [review]
Lgtm

**Pull request merged**